### PR TITLE
Remove nanoFramework.System.Runtime dependency

### PR DESCRIPTION
**Pull request by CoryCharlton**
## Description
Remove nanoFramework.System.Runtime dependency

## Motivation and Context
Resolves new runtime dependency

## How Has This Been Tested?
- Manually

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [X] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [X] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

## Summary by CodeRabbit

- **Chores**
  - Removed outdated dependency on `nanoFramework.System.Runtime` version 1.0.27 to streamline the package management.

### CODE REVIEW

**Comment by jazzman55** [comment]
I believe this would fix the recent error on tests on our CI:

Link failure: some assembly references cannot be resolved!!
Assembly: NFUnitTest (0.0.0.0) needs assembly 'nanoFramework.TestFramework' (2.1.104.0)
Assembly: nanoFramework.TestFramework (2.1.104.0) needs assembly 'nanoFramework.System.Runtime' (1.0.27.0)
Assembly: nanoFramework.UnitTestLauncher (0.0.0.0) needs assembly 'nanoFramework.TestFramework' (2.1.104.0)
Error: a3000000

**Comment by CoryCharlton** [comment]
> I believe this would fix the recent error on tests on our CI:
> 
> Link failure: some assembly references cannot be resolved!! Assembly: NFUnitTest (0.0.0.0) needs assembly 'nanoFramework.TestFramework' (2.1.104.0) Assembly: nanoFramework.TestFramework (2.1.104.0) needs assembly 'nanoFramework.System.Runtime' (1.0.27.0) Assembly: nanoFramework.UnitTestLauncher (0.0.0.0) needs assembly 'nanoFramework.TestFramework' (2.1.104.0) Error: a3000000

This change will definitely fix that error. In the meantime you can downgrade the TestFramework nuget reference OR directly reference nanoFramework.System.Runtime in your test projects.

File: source/TestFramework/NullableAttributes.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// ReSharper disable once CheckNamespace
+namespace System.Diagnostics.CodeAnalysis
```
Comment (torbacz) [inline source/TestFramework/NullableAttributes.cs:5]: Wondering if this file should belong to Test lib?

File: source/TestFramework/CallerArgumentExpressionAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// ReSharper disable once CheckNamespace
+namespace System.Runtime.CompilerServices
```
Comment (torbacz) [inline source/TestFramework/CallerArgumentExpressionAttribute.cs:5]: Wondering if this file should belong to Test lib?

File: poc/TestOfTestFrameworkByReference/Test.cs [hunk 3]
```diff
@@ -263,7 +264,7 @@ private static void ThrowsException()
 
     public class SomethingElse
     {
-        public void NothingReally()
+        public void NothingReally(object value, [CallerArgumentExpression(nameof(value))] string parameter = null)
```
Comment (torbacz) [inline poc/TestOfTestFrameworkByReference/Test.cs:267]: Is this required change? The parameters seems not to be used?

File: source/TestFramework/NullableAttributes.cs (hunk 1 above)
Comment (CoryCharlton) [reply source/TestFramework/NullableAttributes.cs:5]: Yes, these are being copied over from nanoFramework.System.Runtime to resolve the strange running dependency that is breaking tests.

File: source/TestFramework/CallerArgumentExpressionAttribute.cs (hunk 2 above)
Comment (CoryCharlton) [reply source/TestFramework/CallerArgumentExpressionAttribute.cs:5]: Same as the nullable file.

File: poc/TestOfTestFrameworkByReference/Test.cs (hunk 3 above)
Comment (CoryCharlton) [reply poc/TestOfTestFrameworkByReference/Test.cs:267]: This method existed previously to show that methods not attributed with a test attribute with get called.

I added the CallerArgumentExpression to confirm/prove that the internal attributes won't clash with the public versions and that everything works like it should.

**Review by torbacz — APPROVED** [review]

**Comment by jazzman55** [comment]
@josesimoes :)

**Comment by josesimoes** [comment]
> @josesimoes :)

Yes? 😁  If you're asking for this to be merged (and trust me, I understand the need to get this fixes asap) it still needs to be validated against mscorlib tests. The latest changes bring nice improvements, agreed. Still we need to make sure that the test framework doesn't get broken or that we lose fundamental features (such as being able to run/update tests in mscorlib) because of those.

**Comment by jazzman55** [comment]
@josesimoes , TestFramework is actually broken right now and this PR fixes that :)
Reference to nanoFramework.System.Runtime was added to the project, but not to the nuspec (so nuget doesn't install it as dependency). AS result, after upgrading to the latest version TestFramework breaks our CI builds and no tests are executing. I understand it needs to be tested, just bringing your attention ;)

File: poc/TestOfTestFrameworkByReference/NFUnitTestByReference.nfproj [hunk 4]
```diff
@@ -49,6 +50,9 @@
     <Reference Include="nanoFramework.Runtime.Native">
       <HintPath>..\packages\nanoFramework.Runtime.Native.1.6.12\lib\nanoFramework.Runtime.Native.dll</HintPath>
     </Reference>
+    <Reference Include="nanoFramework.System.Runtime">
```
Comment (josesimoes) [inline poc/TestOfTestFrameworkByReference/NFUnitTestByReference.nfproj:53]: This project is referencing the TestFramework, therefore doesn't need to have a reference to the Runtime nuget.

**Comment by CoryCharlton** [comment]
> @josesimoes , TestFramework is actually broken right now and this PR fixes that :) Reference to nanoFramework.System.Runtime was added to the project, but not to the nuspec (so nuget doesn't install it as dependency). AS result, after upgrading to the latest version TestFramework breaks our CI builds and no tests are executing. I understand it needs to be tested, just bringing your attention ;)

For reference there is an open discussion regarding this in Discord: https://discord.com/channels/478725473862549535/1243509599223418933

**Review by josesimoes — APPROVED** [review]
LGTM!

**Pull request merged**